server/render/index.js

```diff
@@ -74,8 +75,40 @@ export function render( element, key = JSON.stringify( element ) ) {
 	//todo: render an error?
 }
 
+function getCacheKey( context ) {
+	if ( ! isSectionIsomorphic( context.store.getState() ) || context.user ) {
+		return false;
+	}
+
+	if ( isEmpty( context.query ) ) {
+		return context.pathname;
+	}
+
+	let queryParams = Object.keys( context.query );
+	const cacheConfiguration = get( getIsomorphicSectionConfiguration( context.store.getState() ), 'cache', null );
+
+	if ( cacheConfiguration && cacheConfiguration.queryParams === true ) {
+		return context.canonicalPath;
+	}
+
+	if ( cacheConfiguration && Array.isArray( cacheConfiguration.queryParams.exclude ) ) {
+		queryParams = difference( queryParams, cacheConfiguration.queryParams.exclude );
+	}
+
+	if ( isEmpty( queryParams ) ) {
+		return context.pathname;
+	}
+
+	if ( cacheConfiguration && isEmpty( difference( queryParams, cacheConfiguration.queryParams.include ) ) ) {
+		return context.pathname + '?' + qs.stringify( pick( context.query, queryParams ) );
```

Comment:
Good stuff. Maybe we can generalize this to absorb the other cases above; by [dropping the `exlude` stuff](https://github.com/Automattic/wp-calypso/pull/17134/files#r132920189), we can maybe treat get the [no-cacheConfiguration case](https://github.com/Automattic/wp-calypso/pull/17134/files#diff-d92d930a21711a960b300aeb231d9056R90) from this clause here with `cacheConfiguration.queryParams.include` set to `[]`.